pkg/gui/controllers/helpers/refresh_helper.go

```diff
@@ -249,6 +249,32 @@ func (self *RefreshHelper) refreshCommits() {
 	wg.Wait()
 }
 
+func (self *RefreshHelper) determineCheckoutBranchName() string {
+	if rebasedBranch := self.c.Git().Status.BranchBeingRebased(); rebasedBranch != "" {
+		// During a rebase we're on a detached head, so cannot determine the
+		// branch name in the usual way. We need to read it from the
+		// ".git/rebase-merge/head-name" file instead.
+		return strings.TrimPrefix(rebasedBranch, "refs/heads/")
+	}
+
+	if bisectedBranch := self.c.Git().Bisect.GetInfo().GetStartSha(); bisectedBranch != "" {
+		// Likewise, when we're bisecting we're on a detached head as well. In
+		// this case we read the branch name from the ".git/BISECT_START" file.
+		return bisectedBranch
+	}
+
+	// In all other cases, get the branch name by asking git what branch is
+	// checked out. Note that if we're on a detached head (for reasons other
+	// than rebasing or bisecting, i.e. it was explicitly checked out), then
+	// this will return its sha.
+	if branchInfo, err := self.c.Git().Branch.CurrentBranchInfo(); err == nil {
```

Comment:
I'm calling `CurrentBranchInfo` here because it exists already. It does more than we need though, so I could add a function that calls `git rev-parse --abbrev-ref HEAD` like you suggested. (`CurrentBranchName()`?)

Reply:
Yep, having that as its own function makes sense

Reply:
See e531aac49d86